Add enabled flag

antedebaas opened the pull request:
I need a flag to check if a config is enabled.

antedebaas committed:
add is active

secit-pl commented:
Please update the documentation before merge.

Also, I think there should be some logic behind this "flag". Please add checking if connection is enabled in methods `getMailbox`, `testConnection` and `tryTestConnection` and add the column "enabled" in the `ValidateConnectionsCommand`.

antedebaas committed:
rename to Enabled and add testing/validating

antedebaas committed:
update documentation

antedebaas commented:
there you go.
renamed it to enabled so the column matches the config

antedebaas commented:
@secit-pl friendly ping

secit-pl commented on Command/ValidateConnectionsCommand.php:
```diff
@@ -107,11 +107,19 @@ private function getRow(ConnectionInterface $connection): array
         }
 
+        if($connection->isEnabled())
+        {
```
Please place opening curly bracket in the same line as if (see Symfony Coding Standard https://symfony.com/doc/current/contributing/code/standards.html#symfony-coding-standards-in-detail)

plus space after "if"

secit-pl commented on Connection/Connection.php:
```diff
@@ -97,6 +97,9 @@ public function getMailbox(): Mailbox
     {
+        if(!$this->isEnabled()) {
+            throw new ConnectionException(array('Mailbox is not enabled'));
+        }
```
Please add new line after closing curly bracket (see: Symfony Coding Standard https://symfony.com/doc/current/contributing/code/standards.html#symfony-coding-standards-in-detail)

secit-pl commented on Connection/Connection.php:
```diff
@@ -117,6 +120,9 @@ public function testConnection(bool $throwExceptions = false): bool
     {
+        if(!$this->isEnabled()) {
+            throw new ConnectionException(array('Mailbox is not enabled'));
+        }
```
Please add new line after closing curly bracket (see: Symfony Coding Standard https://symfony.com/doc/current/contributing/code/standards.html#symfony-coding-standards-in-detail)

secit-pl commented on Connection/Connection.php:
```diff
@@ -117,6 +120,9 @@ public function testConnection(bool $throwExceptions = false): bool
 
     public function tryTestConnection(): void
     {
+        if(!$this->isEnabled()) {
```
Please add space between if and "(" (see: Symfony Coding Standard https://symfony.com/doc/current/contributing/code/standards.html#symfony-coding-standards-in-detail)

secit-pl commented on Command/ValidateConnectionsCommand.php:
```diff
@@ -107,11 +107,19 @@ private function getRow(ConnectionInterface $connection): array
             $result = sprintf('<error>FAILED: %s</error>', $exception->getErrors('last'));
         }
 
+        if($connection->isEnabled())
```
Please add space between if and "(" (see: Symfony Coding Standard https://symfony.com/doc/current/contributing/code/standards.html#symfony-coding-standards-in-detail)

secit-pl commented on Connection/Connection.php:
```diff
@@ -82,6 +88,7 @@ public function getMailbox(): Mailbox
                 $this->password,
                 $this->attachmentsDir,
                 $this->serverEncoding,
+                $this->enabled,
```
As I can see the 6th argument of the [Mailbox::__construct()](https://github.com/barbushin/php-imap/blob/94107fdd1383285459a7f6c2dd2f39e25a1b8373/src/PhpImap/Mailbox.php#L168) is  $trimImapPath, why are you passing enabled to it?

secit-pl commented on Connection/Connection.php:
```diff
@@ -90,6 +97,9 @@ public function getMailbox(): Mailbox
     
     public function testConnection(bool $throwExceptions = false): bool
     {
+        if(!$this->isEnabled()) {
```
Please add space between if and "(" (see: Symfony Coding Standard https://symfony.com/doc/current/contributing/code/standards.html#symfony-coding-standards-in-detail)

secit-pl commented on Connection/Connection.php:
```diff
@@ -90,6 +97,9 @@ public function getMailbox(): Mailbox
     public function testConnection(bool $throwExceptions = false): bool
     {
+        if(!$this->isEnabled()) {
+            throw new ConnectionException(array('Mailbox is not enabled'));
```
testConnection should always return true or false and should not throw error in any case. You should return false here.

secit-pl commented on Connection/Connection.php:
```diff
@@ -110,6 +120,9 @@ public function testConnection(bool $throwExceptions = false): bool
     public function tryTestConnection(): void
     {
+        if(!$this->isEnabled()) {
+            throw new ConnectionException(array('Mailbox is not enabled'));
```
I do not thin that we should thow here ConnectionException as it indicates that the problem is in PhpImap library, not in the bundle. I think we should thow here ErrorException. Additionally please do not use array() notation, please use [] instead.

secit-pl commented on Command/ValidateConnectionsCommand.php:
```diff
@@ -107,11 +107,19 @@ private function getRow(ConnectionInterface $connection): array
 
+        if($connection->isEnabled())
+        {
+            $enabled = "<info>YES</info>";
```
Please use ' instead of "

secit-pl commented on Command/ValidateConnectionsCommand.php:
```diff
@@ -107,11 +107,19 @@ private function getRow(ConnectionInterface $connection): array
+        {
+            $enabled = "<info>YES</info>";
+        } else {
+            $enabled = "<error>NO</error>";
```
Please use ' instead of "

secit-pl commented on DependencyInjection/Configuration.php:
```diff
@@ -55,6 +55,8 @@ public function getConfigTreeBuilder(): TreeBuilder
                                 ->cannotBeEmpty()
                                 ->defaultValue('UTF-8')
                             ->end()
+                            ->scalarNode('enabled')
```
Please add default value as in create_attachments_dir_if_not_exists: "->defaultTrue()" to avoid BC breaks.

secit-pl reviewed (changes requested):
You won't like me :) Please see my comments. Most of them are related to the coding standard.

antedebaas committed:
process comments

secit-pl commented:
Changes looks ok. One thought before merge, shouldn't there go some logic behind the isEnabled()? At this moment the flag doesn't do anything. Maybe we should block possibility to get the mailbox (getMailbox() -> if (!$connection->isEnabled()) throw \ErrorException('connection disabled')) and do not initiate connection on the test command for disabled connections? WDYT?

antedebaas committed:
dont allow the mailbox to be opend when disabled

antedebaas commented:
I agree;
Originally I had the idea of using the flag on my project side. But for this it makes more sense to add logic to it here so I added code for it.
i also modified the validate-connection command not to try and open the mailbox when its disabled

antedebaas committed:
make it a connection exception

secit-pl commented:
Merged. Thank you for your contribution!
